lib/concurrent/edge/future.rb

```diff
@@ -100,25 +128,26 @@ def initialize(promise, default_executor = :io)
         ensure_ivar_visibility!
       end
 
+      # @return [:pending, :completed]
       def state
-        @State.get
+        @State.get.to_sym
```

Comment:
Why do we opt to return a sym here? Could this potentially be a source of a memor